docs/storage-surface.rst

```diff
@@ -0,0 +1,673 @@
+   True
+
+To delete a blob
+
+.. code-block:: python
+
+   >>> blob.delete()
+
+To generate a signed URL for temporary privileged access to the
+contents of a blob
+
+.. code-block:: python
+
+   >>> expiration_seconds = 600
+   >>> signed_url = blob.generate_signed_url(expiration_seconds)
+
+To make updates to the blob use
+:meth:`Blob.patch() <gcloud.storage.blob.Blob.patch>`
+
+.. code-block:: python
+
+   >>> blob.versioning_enabled = True
+   >>> blob.patch()
+
+If there are no updates to send, an exception will occur
+
+.. code-block:: python
+
+   >>> blob.patch()
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   ValueError: No updates to send.
+
+In total, the properties that can be updated are
+
+.. code-block:: python
+
+   >>> blob.cache_control = 'private, max-age=0, no-cache'
+   >>> blob.content_disposition = 'Attachment; filename=example.html'
+   >>> blob.content_encoding = 'gzip'
+   >>> blob.content_language = 'en-US'
+   >>> blob.content_type = 'text/plain'
+   >>> blob.crc32c = u'z8SuHQ=='  # crc32-c of "foo"
+   >>> blob.md5_hash = u'rL0Y20zC+Fzt72VPzMSk2A=='  # md5 of "foo"
+   >>> blob.metadata = {'foo': 'bar', 'baz': 'qux'}
+
+.. note::
+  **BREAKING THE FOURTH WALL**: Why are ``crc32c`` and ``md5_hash`` writable?
+
+In general, many of these properties are optional and will not need to be
+used (or changed from the defaults).
+
+In addition, a blob has several read-only properties
+
+.. code-block:: python
+
+   >>> blob.bucket
+   <Bucket: bucket-name>
+   >>> blob.component_count
+   1
+   >>> blob.etag
+   u'CNiOr665xcQCEAE='
+   >>> blob.generation
+   12345L
+   >>> blob.id
+   u'bucket-name/blob-name/12345'
+   >>> blob.media_link
+   u'https://www.googleapis.com/download/storage/v1/b/bucket-name/o/blob-name?generation=12345&alt=media'
+   >>> blob.metageneration
+   1L
+   >>> blob.name
+   'blob-name'
+   >>> blob.owner
+   {u'entity': u'user-01234',
+    u'entityId': u'01234'}
+   >>> blob.self_link
+   u'https://www.googleapis.com/storage/v1/b/bucket-name/o/blob-name'
+   >>> blob.size
+   3L
+   >>> blob.storage_class
+   u'STANDARD'
+   >>> print blob.time_deleted
+   None
+   >>> blob.updated
+   u'2015-01-01T12:00:00.000Z'
+
+See `objects`_ specification for more details.
+
+Other data -- namely `access control`_ data -- is associated with blobs,
+but is handled through ``Blob.acl``.
+
+.. _objects: https://cloud.google.com/storage/docs/json_api/v1/objects
+
+Working with Blob Data
+----------------------
+
+The most important use of a blob is not accessing and updating the metadata,
+it is storing data in the cloud (hence Cloud Storage).
+
+To upload string data into a blob
+
+  .. code-block:: python
+
+   >>> blob.upload_from_string('foo')
+
+If the data has a known content-type, set it on the blob before
+uploading:
+
+  .. code-block:: python
+
+   >>> blob.content_type = 'application/zip'
+   >>> blob.upload_from_string('foo')
+
+To upload instead from a file
```

Comment:
It has been spooled off to a tempfile (e.g., by the framework), and is available (easily) only as a file-handle.  For download, it would be nice to *get* the fileh-andle back, to return as the `app_iter` in a WSGI response.